src/Sylius/Bundle/AssociationBundle/Behat/ProductAssociationContext.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\AssociationBundle\Behat;
+
+use Behat\Gherkin\Node\TableNode;
+use Sylius\Bundle\ResourceBundle\Behat\DefaultContext;
+use Sylius\Component\Association\Model\AssociationType;
+use Sylius\Component\Association\Model\AssociationTypeInterface;
+use Sylius\Component\Product\Model\ProductAssociation;
+use Sylius\Component\Product\Model\ProductInterface;
+
+/**
+ * @author Leszek Prabucki <leszek.prabucki@gmail.com>
+ */
+class ProductAssociationContext extends DefaultContext
+{
+    /**
+     * @var ProductInterface
+     */
+    private $currentProduct;
+
+    /**
+     * @When I create :typeName association type
+     */
+    public function iCreateAssociationType($typeName)
+    {
+        $this->fillField('Name', $typeName);
+        $this->pressButton('Create');
+        $this->assertSession()->pageTextContains('Association type has been successfully created.');
+    }
+
+    /**
+     * @Then I should be able to add :typeName associations to every product
+     */
+    public function iShouldBeAbleToAddAssociationsToEveryProduct($typeName)
```

Comment:
It's a quite weird step, usually `Then` clauses shouldn't modify the state of application, but check if it fulfills the expectations. IMHO we can just go on product association adding page anc check whether there is option with given type available.